cogeo_mosaic/backends/base.py

```diff
@@ -61,6 +63,59 @@ def quadkey_zoom(self) -> int:
     def write(self):
         """Upload new MosaicJSON to backend."""
 
-    @abc.abstractmethod
-    def update(self):
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update quadkey list."""
+        self.mosaic_def.tiles[quadkey] = dataset
+
+    def _update_metadata(self, bounds: List[float], version: str):
+        """Update bounds and center."""
+        self.mosaic_def.version = version
+        self.mosaic_def.bounds = bounds
+        self.mosaic_def.center = (
+            (bounds[0] + bounds[2]) / 2,
+            (bounds[1] + bounds[3]) / 2,
+            self.mosaic_def.minzoom,
+        )
+
+    def update(
+        self,
+        features: Sequence[Dict],
+        add_first: bool = True,
+        mosaic_class=MosaicJSON,
```

Comment:
I'm adding `mosaic_class` as an option but not sure it is needed ? How does one use a custom mosaic class (e.g custom filter method) with backends? 

Backends have `MosaicJSON` hardcoded https://github.com/developmentseed/cogeo-mosaic/blob/MosaicClass/cogeo_mosaic/backends/base.py#L15, should this be passed as an options ? cc @geospatial-jeff @kylebarron

Reply:
Thats a good point, I hadn't thought of that.  I think the options are (1) have the user monkeypatch the `_filter` class method or (2) pass a custom `MosaicJSON` as you mentioned.

Reply:
alight I think https://github.com/developmentseed/cogeo-mosaic/pull/67 would make it work. 

MosaicJSON class shouldn't really need to be customized, only the `asset_filter`and the `accessor` need to be.

With #67, you can now pass the custom `asset_filter` directly to the update method.

Reply:
That being said I think it's still not perfect because you don't have the ability to apply filter on the whole stack of data (new data and old data). To be able to do this you'll need a customBackend with a custom `Update` method.

Reply:
maybe we can turn https://github.com/developmentseed/cogeo-mosaic/pull/65/files#diff-79db4f358a4b5404256fcb377a494e21R96 into another `Callable` 🤔